sylt-compiler/src/typechecker.rs

```diff
@@ -1400,34 +1410,45 @@ impl TypeChecker {
             .constraints
             .clone()
             .iter()
-            .map(|con| match &con {
-                Constraint::Add(x) => Constraint::Add(self.inner_copy(*x, seen)),
-                Constraint::Sub(x) => Constraint::Sub(self.inner_copy(*x, seen)),
-                Constraint::Mul(x) => Constraint::Mul(self.inner_copy(*x, seen)),
-                Constraint::Div(x) => Constraint::Div(self.inner_copy(*x, seen)),
-                Constraint::Equ(x) => Constraint::Equ(self.inner_copy(*x, seen)),
-                Constraint::Cmp(x) => Constraint::Cmp(self.inner_copy(*x, seen)),
-                Constraint::CmpEqu(x) => Constraint::CmpEqu(self.inner_copy(*x, seen)),
-                Constraint::Neg => Constraint::Neg,
-                Constraint::Indexes(x) => Constraint::Indexes(self.inner_copy(*x, seen)),
-                Constraint::IndexedBy(x) => Constraint::IndexedBy(self.inner_copy(*x, seen)),
-                Constraint::IndexingGives(x) => {
-                    Constraint::IndexingGives(self.inner_copy(*x, seen))
-                }
-                Constraint::GivenByIndex(x) => Constraint::GivenByIndex(self.inner_copy(*x, seen)),
-                Constraint::ConstantIndex(i, x) => {
-                    Constraint::ConstantIndex(*i, self.inner_copy(*x, seen))
-                }
-                Constraint::Field(f, x) => Constraint::Field(f.clone(), self.inner_copy(*x, seen)),
-                Constraint::Num => Constraint::Num,
-                Constraint::Container => Constraint::Container,
-                Constraint::SameContainer(x) => {
-                    Constraint::SameContainer(self.inner_copy(*x, seen))
-                }
-                Constraint::Contains(x) => Constraint::Contains(self.inner_copy(*x, seen)),
-                Constraint::IsContainedIn(x) => {
-                    Constraint::IsContainedIn(self.inner_copy(*x, seen))
-                }
+            .map(|(con, span)| {
+                (
+                    match &con {
+                        Constraint::Add(x) => Constraint::Add(self.inner_copy(*x, seen)),
+                        Constraint::Sub(x) => Constraint::Sub(self.inner_copy(*x, seen)),
+                        Constraint::Mul(x) => Constraint::Mul(self.inner_copy(*x, seen)),
+                        Constraint::Div(x) => Constraint::Div(self.inner_copy(*x, seen)),
+                        Constraint::Equ(x) => Constraint::Equ(self.inner_copy(*x, seen)),
+                        Constraint::Cmp(x) => Constraint::Cmp(self.inner_copy(*x, seen)),
+                        Constraint::CmpEqu(x) => Constraint::CmpEqu(self.inner_copy(*x, seen)),
+                        Constraint::Neg => Constraint::Neg,
+                        Constraint::Indexes(x) => Constraint::Indexes(self.inner_copy(*x, seen)),
+                        Constraint::IndexedBy(x) => {
+                            Constraint::IndexedBy(self.inner_copy(*x, seen))
```

Comment:
We should have some test with unknown dicts and lists.

Reply:
Feel free to write those tests - these should not be a part of this PR though. This was just a formatting change.